Source/Core/Core/AchievementManager.cpp

```diff
@@ -633,6 +633,13 @@ std::recursive_mutex* AchievementManager::GetLock()
   return &m_lock;
 }
 
+bool AchievementManager::IsHardcoreModeActive()
+{
+  std::lock_guard lg{m_lock};
+  return Config::Get(Config::RA_HARDCORE_ENABLED) && IsGameLoaded() &&
+         (m_runtime.trigger_count + m_runtime.lboard_count > 0);
+}
```

Comment:
I'm aware this logic is wrong, I fell asleep halfway through writing it and forgot it was unfinished when I woke up. Will fix at next opportunity.